### PR TITLE
Add alert preview

### LINKED ISSUE #7: Adding a preview?
It would be nice to see a preview of the notice while editing the block.

If it's ok, I can implement it.

### PR DESCRIPTION
**Pull request by mlocati**
Closes #7

Preview:

preview

### CODE REVIEW

File: blocks/pure_cookies_notice/controller.php [hunk 1]
```diff
@@ -198,36 +204,41 @@ private function shouldShowAgreement()
...
-            $lines[] = "\tborder-color: {$this->textColor};";
+        $lines[] = "#pure-cookies-notice-{$controller->bID} .pure-cookies-notice-close-button {";
+        if (!empty($controller->textColor)) {
+            $lines[] = "\tcolor: {$controller->textColor};";
+            $lines[] = "\tborder-color: {$controller->textColor};";
         }
         $lines[] = '}';
         $lines[] = '</style>';
 
         return implode("\n", $lines);
```
Comment (guyasyou) [inline blocks/pure_cookies_notice/controller.php:234]: maybe need use PHP_EOL?

Comment (mlocati) [reply blocks/pure_cookies_notice/controller.php:234]: Well, it's the same, both will work.
BTW all the other code uses the Unix line ending (`\n`), so I'd prefer `\n`

**Comment by guyasyou** [comment]
chrome_2019-01-14_23-07-17
I have error in packages/pure_cookies_notice/blocks/pure_cookies_notice/controller.php::344
Concrete5  8.4.4

**Comment by mlocati** [comment]
I should have fixed the error: now the code should run for concrete5 8.1+ (as specified in `$appVersionRequired` of package controller).

File: blocks/pure_cookies_notice/controller.php [hunk 2]
```diff
@@ -341,7 +341,12 @@ public function action_generate_preview()
             }
             throw new Exception($token->getErrorMessage());
         }
-        $blockType = $this->app->make(EntityManagerInterface::class)->find(BlockType::class, $this->getBlockTypeID());
+        $em = $this->app->make(EntityManagerInterface::class);
+        if (method_exists($this, 'getBlockTypeID')) {
+            $blockType = $em->find(BlockType::class, $this->getBlockTypeID());
+        } else {
+            $blockType = $em->getRepository(BlockType::class)->findOneBy(['btHandle' => $this->btHandle]);
```
Comment (guyasyou) [inline blocks/pure_cookies_notice/controller.php]: We can't use this always?

Comment (guyasyou) [inline blocks/pure_cookies_notice/controller.php:348]: We can't use this always?

**Comment by mlocati** [comment]
[In the new code](https://github.com/guyasyou/Pure-Cookies-Notice/pull/9/commits/9ce35f7bd709b7dd6898a4c5d54830859aae5a84#diff-4a3d26cb98c7308a0f02aebc5004cacbR345), we could just use the `$this->btHandle` approach, but I prefer the first approach (`$this->getBlockTypeID()`) because it's faster, and I prefer using getters/setters instead properties.
BTW we need to wait for the next concrete5 version in order to use `getBlockTypeID` (it has been introduced in [#6976](https://github.com/concrete5/concrete5/pull/6976)).

**Review by guyasyou — APPROVED** [review]

**Comment by mlocati** [comment]
Thanks!

**Comment by guyasyou** [comment]
I think to rollback this. Why we need to displaying real data?
Simply show how the notification looks in the selected colors in the color's tab. It is just JS, without changes in controller

**Pull request merged**